code/modules/goonchat/browserassets/js/browserOutput.js

```diff
@@ -95,23 +85,66 @@ if (typeof String.prototype.trim !== 'function') {
 	};
 }
 
-//Shit fucking piece of crap that doesn't work god fuckin damn it
-function linkify(text) {
-	var rex = /((?:<a|<iframe|<img)(?:.*?(?:src="|href=").*?))?(?:(?:https?:\/\/)|(?:www\.))+(?:[^ ]*?\.[^ ]*?)+[-A-Za-z0-9+&@#\/%?=~_|$!:,.;]+/ig;
-	return text.replace(rex, function ($0, $1) {
-		if(/^https?:\/\/.+/i.test($0)) {
-			return $1 ? $0: '<a href="'+$0+'">'+$0+'</a>';
-		}
-		else {
-			return $1 ? $0: '<a href="http://'+$0+'">'+$0+'</a>';
-		}
-	});
+//This works, don't fuck with it k, or I'll break your fingers
+function linkify(message) {
+        /* See stack overflow for the regex: https://stackoverflow.com/questions/6038061/regular-expression-to-find-urls-within-a-string 
+		 * it gets the domain, protocol, and query string as three match groups
+		 * 
+		 * Note that we only replace links within spans containing the linkify span text, so we don't replace stuff we need to not fuck up
+		 * like the src url for goonchat images, or other assets and so forth
+		 *
+		 * basically only ooc chat span messages have this span class, why also have a flag you then ask?
+		 * well, if we can avoid doing this function call at all, that's free performance from not doing
+		 * regexes over text
+		 * 
+		 * Hey why not do a proper js url parser? well, because I don't feel like dealing with possible ie8/9 backward compat issues, feel free to try
+         */
+		var rex = /(?:(http|byond|https):\/\/)?([\w_-]+(?:(?:\.[\w_-]+)+))([\w.,@?^=%&:/~+#-]*[\w@?^=%&/~+#-])?/ig;
```

Comment:
this should be a global

Reply:
We have a global but it's outdated and barely used
https://github.com/tgstation/tgstation/blob/4f7a382cdd4e4d0b2fda1784b59f7dee3e32e15d/code/_globalvars/regexes.dm#L4

Reply:
that's also in DM, which isn't very helpful for this javascript, this was just a note to myself to move the regex up to the top level and init it in the goonchat startup function

Reply:
whoops forgot it was js